specifyweb/specify/views.py

```diff
@@ -550,6 +558,7 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
                 if table_name in dependant_table_names:
                     # Note: need to handle case where deletion throws error because it is referenced my other records
                     try:
+                        clean_fields_pre_delete(obj)
                         obj.delete()
```

Comment:
> Doing this, preserves the attachment location since the front-end sends data for dependents.

https://github.com/specify/specify7/pull/4022#issuecomment-1723921603

Would it be worth adding a note of this behavior here? 
While it is known to us, this might be initially confusing to future developers who stumble upon this code (assuming the code remains untouched). 
They might stumble upon this in the future and think it is a potential bug without making the connection that the frontend sends/overrides the data for dependent resources.